### ISSUE TITLE
Signal errors with return codes?

### ISSUE THREAD

**Issue by ryan-williams** [Blosc/c-blosc#239]
While decompressing via JBlosc, I was hitting [a `-1` return value from because my "dest" buffer wasn't big enough](https://github.com/Blosc/c-blosc/blob/a60e30bf371b50dff22a50e224e95752ee41dfb5/blosc/blosc.c#L1446).

I'm responding to all `-1`s by exponentially increasing my buffer size until I hit 2GB-1, at which point I assume continued failure means the data is corrupt, but it would be nice to differentiate these cases via the return value.

**Comment by FrancescAlted** [comment]
Yes, I'm +1 on this, although the error codes would need a more complete overhauling in general.  Help in this area is welcome.

**Comment by ryan-williams** [comment]
Cool, I was kind of imagining that this would be considered a "breaking change" and not really feasible, but I if there were explicitly no guarantees about return codes previously, other than e.g. "negative means error", then it's not a big deal?

Would the process basically be to replace every `return -1` with a unique integer, and document it?
